fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -811,14 +812,27 @@ private List<IndexDef> getIndexDefs(List<StarRocksParser.IndexDescContext> index
             List<Identifier> columnList = visit(context.identifierList().identifier(), Identifier.class);
             String comment =
                     context.comment() != null ? ((StringLiteral) visit(context.comment())).getStringValue() : null;
+
             final IndexDef indexDef =
                     new IndexDef(indexName, columnList.stream().map(Identifier::getValue).collect(toList()),
-                            IndexDef.IndexType.BITMAP, comment, createPos(context));
+                            getIndexType(context.indexType()), comment, getProperties(context.properties()), createPos(context));
             indexDefList.add(indexDef);
         }
         return indexDefList;
     }
 
+    private IndexDef.IndexType getIndexType(IndexTypeContext indexTypeContext) {
```

Comment:
It is better to move this function into IndexType class.

Reply:
Done